plugins/nv-filters/nvidia-videofx-filter.c

```diff
@@ -149,73 +216,252 @@ static void nv_greenscreen_filter_actual_destroy(void *data)
+	case S_FX_BG_BLUR:
+		vfxErr = NvVFX_CreateEffect(NVVFX_FX_GREEN_SCREEN,
+					    &filter->handle);
+		if (NVCV_SUCCESS != vfxErr)
+			log_nverror_destroy(filter, vfxErr);
+		vfxErr = NvVFX_CreateEffect(NVVFX_FX_BGBLUR,
+					    &filter->handle_blur);
+		break;
+	default:
+		return false;
+	}
+	if (NVCV_SUCCESS != vfxErr)
+		log_nverror_destroy(filter, vfxErr);
+		/* debug */
+#ifdef _DEBUG
+	const char *info;
+	vfxErr = NvVFX_GetString(filter->handle_blur, NVVFX_INFO, &info);
+	blog(LOG_DEBUG, "blur fx settings /n/%s", info);
+#endif
+	/* 2. Set models path & initialize CudaStream */
+	if (id == S_FX_AIGS || id == S_FX_BG_BLUR) {
+		char buffer[MAX_PATH];
+		char modelDir[MAX_PATH];
+		nvvfx_get_sdk_path(buffer, MAX_PATH);
+		size_t max_len = sizeof(buffer) / sizeof(char);
+		snprintf(modelDir, max_len, "%s\\models", buffer);
+		vfxErr = NvVFX_SetString(filter->handle, NVVFX_MODEL_DIRECTORY,
+					 modelDir);
+		vfxErr = NvVFX_CudaStreamCreate(&filter->stream);
+		if (NVCV_SUCCESS != vfxErr)
+			log_nverror_destroy(filter, vfxErr);
+		vfxErr = NvVFX_SetCudaStream(filter->handle, NVVFX_CUDA_STREAM,
+					     filter->stream);
+		if (NVCV_SUCCESS != vfxErr)
+			log_nverror_destroy(filter, vfxErr);
+	}
+	if (id == S_FX_BLUR || id == S_FX_BG_BLUR) {
+		vfxErr = NvVFX_CudaStreamCreate(&filter->stream_blur);
+		if (NVCV_SUCCESS != vfxErr)
+			log_nverror_destroy(filter, vfxErr);
+		vfxErr = NvVFX_SetCudaStream(filter->handle_blur,
+					     NVVFX_CUDA_STREAM,
+					     filter->stream_blur);
+		if (NVCV_SUCCESS != vfxErr)
+			log_nverror_destroy(filter, vfxErr);
+	}
+	return true;
+}
+
+static void *nvvfx_filter_create(obs_data_t *settings, obs_source_t *context,
+				 enum nvvfx_fx_id id)
+{
+	struct nvvfx_data *filter =
+		(struct nvvfx_data *)bzalloc(sizeof(*filter));
+	if (!nvvfx_loaded) {
+		nvvfx_filter_destroy(filter);
+		return NULL;
+	}
+
+	NvCV_Status vfxErr;
+	filter->context = context;
+	filter->mode = -1; // should be 0 or 1; -1 triggers an update
+	filter->images_allocated = false;
+	filter->processed_frame = true; // start processing when false
+	filter->width = 0;
+	filter->height = 0;
+	filter->initial_render = false;
+	os_atomic_set_bool(&filter->processing_stop, false);
+	filter->handler = NULL;
+	filter->processing_interval = 1;
+	filter->processing_counter = 0;
+	// set nvvfx_fx_id
+	filter->filter_id = id;
+	// blur specific
+	filter->strength = -1;
+
+#ifdef _DEBUG
+	/* sanity check of sdk version */
+	if (NvVFX_GetVersion(&filter->version) == NVCV_SUCCESS) {
+		uint8_t major = (filter->version >> 24) & 0xff;
+		uint8_t minor = (filter->version >> 16) & 0x00ff;
+		uint8_t build = (filter->version >> 8) & 0x0000ff;
+		uint8_t revision = (filter->version >> 0) & 0x000000ff;
+		nvvfx_new_sdk = filter->version >= MIN_VFX_SDK_VERSION &&
+				nvvfx_new_sdk;
+	}
+#endif
+	/* 1. Create FX */
+	/* 2. Set models path & initialize CudaStream */
+	if (!nvvfx_filter_create_internal(filter))
+		return NULL;
+
+	/* 3. Load effect. */
```

Comment:
```suggestion
	/* 3. Load effect */
```

Generally "bullet points" don't end in final punctuation, unless the bullet point text requires non-trivial punctuation (multiple sentences, or other punctuation).

Reply:
thanks, point noted